samtranslator/model/appsync.py

```diff
@@ -0,0 +1,24 @@
+from typing import Optional, Dict, Any, List
+from typing_extensions import TypedDict
+
+from samtranslator.model import PropertyType, Resource
+from samtranslator.model.types import IS_DICT, is_type, list_of, IS_STR
+
+
+class Auth(TypedDict, total=False):
+    Type: str
+
+
+class GraphQLApi(Resource):
+    resource_type = "AWS::AppSync::GraphQLApi"
+    property_types = {
+        "Name": PropertyType(True, IS_STR),
+        "Tags": PropertyType(False, list_of(IS_DICT)),
+        "XrayEnabled": PropertyType(False, is_type(bool)),
+        "AuthenticationType": PropertyType(True, IS_STR),
```

Comment:
This doesn't look complete to me. Any reason we aren't adding all the properties? https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html

Reply:
It's not. complete obviously. The branch is work-in-progress

Reply:
This is just a very high level mapping to start with the most basic necessary properties. I think it's best to move slowly with this as we may refactor other parts of the design, but these core properties will certainly remain.

Also, some of the configuration properties can get a bit complex so I think it's better to take it slow here and add functionality in small pieces.

Reply:
Makes sense. Thanks for the clarifying.